src/wallet/coincontrol.h

```diff
@@ -38,6 +39,8 @@ class CCoinControl
     //! If true, the selection process can add extra unselected inputs from the wallet
     //! while requires all selected inputs be used
     bool m_allow_other_inputs = false;
+    //! Filter inputs by type
+    std::unordered_set<OutputType> m_filter_inputs;
```

Comment:
Small idea, what about using an optional bit-field here?
So you later can skip coins with a fast:
```c++
if (m_filter_inputs && !(*m_filter_inputs & coin_type)) continue;
```

Reply:
great idea, I pushed it
`git diff 0a9d345 89ea8fc`